src/payment/strategies/paypal-commerce/paypal-commerce-payment-processor.ts

```diff
@@ -143,6 +133,19 @@ export default class PaypalCommercePaymentProcessor {
         return this._hostedFields.submit(options);
     }
 
+    validateHostedForm(): { isValid: boolean; fields: PaypalCommerceHostedFieldsState['fields'] } {
```

Comment:
Correct me if I'm wrong, but it looks like this method doesn't actually validate the form. It only gets the result of the validation result that's done previously (i.e.: when a user type into the form) or the initial state. Does PayPal offer a method to actually do the validation? If not, it'd be good to rename to the method, i.e.: `getValidationState` so it's clear to other people reading the code that it's a getter rather than an action.

Another thing to check is, what happens when a user clicks on the submit button without making any changes to the form? Are the initial values of `isValid` false?

Reply:
Yes, you are right. I will rename method. Thank you!

PayPal doesn't offer a method to actually do the validation. You can see example on the screenshot.
And  the initial values are `isValid=false` when a user clicks on the submit button without making any changes

<img width="769" alt="Screenshot 2020-10-16 at 10 47 04" src="https://user-images.githubusercontent.com/32959076/96227611-069a7500-0f9d-11eb-861b-ed7bb072d4d3.png">